### PR TITLE
feat: Adds YAML Linting

### PR DESCRIPTION
**Pull request by ElvenSpellmaker**

### CODE REVIEW

File: build/azDevOps/packages-amido-stacks-configuration.yml [hunk 1]
```diff
@@ -20,7 +19,15 @@ variables:
...
-
+  # SelfConfig
+  - name: self_repo
+    value: stacks-dotnet-packages-configuration
+  - name: self_repo_dir
+    value: "$(Agent.BuildDirectory)/s/$(self_repo)"
+  - name: self_pipeline_repo
+    value: "$(Agent.BuildDirectory)/s/stacks-pipeline-templates"
+  - name: self_pipeline_scripts_dir
+    value: "$(self_pipeline_repo)/scripts"
```
Comment (chris31389) [inline build/azDevOps/packages-amido-stacks-configuration.yml:30]: As this folder doesn't exist in the repo, how is it used by the yaml validation job?

Comment (chris31389) [reply build/azDevOps/packages-amido-stacks-configuration.yml:30]: cancel this, i've just seen it is using the `stacks-pipeline-templates` repo

**Review by chris31389 — COMMENTED** [review]
I'm curious whether its possible to move the Validation step into the `stacks-pipeline-templates` repo and use it from there.  If that is the case, I'd like to see if the config file could be included in the template repo and set as a default with the option of overriding the config file.   Other than that curiousity, i'm happy with these changes.

File: build/azDevOps/packages-amido-stacks-configuration.yml [hunk 2]
```diff
@@ -34,14 +41,20 @@ variables:
...
 trigger:
   branches:
     include:
-      - main
-      - feature/*
-      - bug*
+    - 'main'
+  paths:
+    include:
+    - '*'
```
Comment (chris31389) [inline build/azDevOps/packages-amido-stacks-configuration.yml:57]: Any reason you've added this back in?  I believe this is the default settings.

Comment (ElvenSpellmaker) [reply build/azDevOps/packages-amido-stacks-configuration.yml:57]: There were some odd branches specified before, I've set it back to defaults here.

File: build/azDevOps/templates/steps/build/test-validate-yaml.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+parameters:
```
[the remark this answers is not in the document]
Comment (ElvenSpellmaker) [reply build/azDevOps/templates/steps/build/test-validate-yaml.yml:1]: There's for and against to move it into the pipeline repo, stacks-java has it in its own repo, so far there's not a template file which calls bash scripts in the pipeline repo.

File: build/azDevOps/packages-amido-stacks-configuration.yml (hunk 2 above)
Comment (chris31389) [reply build/azDevOps/packages-amido-stacks-configuration.yml:57]: the branches part I get, but this is about the paths. 

> If you don't set path filters, then the root folder of the repo is implicitly included by default. 

https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#ci-triggers

File: build/azDevOps/templates/steps/build/test-validate-yaml.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (chris31389) [reply build/azDevOps/templates/steps/build/test-validate-yaml.yml:1]: https://github.com/amido/stacks-pipeline-templates/blob/feature/cycle4/azDevOps/azure/templates/v2/steps/build-pack-test-dotnet.yml

[the remark this answers is not in the document]
Comment (chris31389) [reply build/azDevOps/templates/steps/build/test-validate-yaml.yml:1]: I think this is exactly why the shared template repo was created, for sharing common scripts across repo's.

File: build/azDevOps/packages-amido-stacks-configuration.yml (hunk 2 above)
Comment (ElvenSpellmaker) [reply build/azDevOps/packages-amido-stacks-configuration.yml:57]: Yeah, I'm just setting the default, being explicit, it's how we've gone it everywhere else.

**Review by chris31389 — APPROVED** [review]

**Pull request merged**